pgml-extension/src/bindings/transformers.py

```diff
@@ -110,22 +110,38 @@ def transform(task, args, inputs):
     return json.dumps(pipe(inputs, **args), cls=NumpyJSONEncoder)
 
 
-def embed(transformer, text, kwargs):
+def embed(transformer, inputs, kwargs):
+    
+    try:
+        inputs = json.loads(inputs)
+    except json.decoder.JSONDecodeError:
+        pass      
+        
     kwargs = json.loads(kwargs)
     ensure_device(kwargs)
     instructor = transformer.startswith("hkunlp/instructor")
+    
     if instructor:
         klass = INSTRUCTOR
-        text = [[kwargs.pop("instruction"), text]]
+        if isinstance(inputs, str):
+            inputs = [[kwargs.pop("instruction"), inputs]]
+            
+        else:
+            texts_with_instructions = []
+            instruction = kwargs.pop("instruction")
```

Comment:
Seems like it may be common to have multiple instructions with multiple inputs? Hmm, in that case I'm not sure we have a nice way to structure the args... we can leave that for some future work.

Reply:
Okay I'll leave it as it is.